### PR TITLE
[RTL] Create new 'comb' dialect and move all the RTL combinational ops into it

### PR DESCRIPTION
**Pull request by teqdruid**
This is step 1 to #523. I put `circt::comb` in the `circt::rtl` namespace so we could do most of the C++ changes in a subsequent PR.

### CODE REVIEW

**Comment by teqdruid** [comment]
I already moved them, but I'm considering moving `BitcastOp` and `ConstantOp` back into the RTL dialect. I could see arguments either way, but the same arguments apply to ops like `ArraySliceOp`, which I've kept in the RTL dialect. Thoughts?

**Comment by teqdruid** [comment]
I haven't had a chance to review this PR myself, so it's just a draft.

File: include/circt/Dialect/Comb/Comb.td [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                          "::mlir::IntegerType">;
+
+// Type constraint that indicates that an operand/result may only be a valid,
+// known, non-directional type.
+def RTLValueType : Type<CPred<"rtl::isRTLValueType($_self)">,
+                        "a known primitive element">;
+
+// Type constraint that indicates that an operand/result may only be a valid
+// non-directional type.
+def RTLNonInOutType : Type<CPred<"!rtl::hasRTLInOutType($_self)">,
```
Comment (teqdruid) [inline include/circt/Dialect/Comb/Comb.td:46]: Do we need this? Presumably there's going to be any number of types which don't make sense but which the `comb` dialect shouldn't need to know about.

Comment (lattner) [reply include/circt/Dialect/Comb/Comb.td:46]: Ok, I'm fine with dropping RTLNonInOutType if you don't want to carry it forward, but please don't conflate such a design change with a mechanical patch like this.

**Review by lattner — APPROVED** [review]
This looks good to me, I agree that constant/bitcast should go with comb.

**Comment by teqdruid** [comment]
The reason I was considering moving them was having them in `comb` introduces a dependence on `comb` from `rtl`. RTL uses `ConstantOp` in one of the builders and `bitcast` isn't used in any `rtl` code, but it only makes sense in the context of RTL types.

**Comment by teqdruid** [comment]
Also, `bitcast` needs `getBitWidth` in the RTL dialect and it wouldn't make sense to bring that function over to the `comb` dialect.

**Comment by lattner** [comment]
Awesome thank you for landing this!  This removed the references to it, but I don't think this actually deleted `RTLStatements.td` and friends.  Can you nuke those when you get a chance?

**Comment by lattner** [comment]
by nuke, I mean "git rm RTLStatements.td"

**Comment by teqdruid** [comment]
RTLStatements.td contains mostly Ops which we said should be in the SV dialect and one which should go in RTLAggregrates.td. I did 'rm' RTLCombinatorial.td in this PR which is the only file which contained combinational ops.

It should be noted that #523 is not quite done yet as we have yet to update the rtl/sv rationale document and possibly some clean up some other dust.

**Comment by lattner** [comment]
`RTLStatements.td` is dead code, and should just be deleted.  Those ops have already been moved

**Pull request merged**